src/uutf.ml

```diff
@@ -701,7 +707,7 @@ module String = struct
   type 'a folder =
     'a -> int -> [ `Uchar of uchar | `Malformed of string ] -> 'a
 
-  let fold_utf_8 f acc s =
+  let fold_utf_8 f acc s ?(pos=0) ?len () =
```

Comment:
You should put the labels before `f` and forget about the `()`. That's the way to do it and it will maximise the chances that code will remain backward compatible.